src/jshint.js

```diff
@@ -2767,6 +2767,9 @@ var JSHINT = (function() {
           break;
         }
       } else {
+        if (state.option.trailingcomma && state.option.inES5(true)) {
```

Comment:
That check was accommodate [IE < 9 trailing comma bugs](http://stackoverflow.com/a/11306770/1468130). I guess it isn't really a matter of the ES version, instead whether old IE bugs need to be taken into account. Old IE == ES3 to me, I guess.